src/main/java/org/broadinstitute/consent/http/service/UserService.java

```
@@ -391,4 +395,26 @@ private Boolean checkForValidInstitution(Integer institutionId) {
         return Objects.nonNull(existingId) && existingId > 0;
     }
 
+    public User findOrCreateUser(AuthUser authUser) {
+        User user;
+        // Ensure that the user is a registered DUOS user
+        try {
+            user = userDAO.findUserByEmail(authUser.getEmail());
+        } catch (NotFoundException nfe) {
+            User newUser = new User();
+            newUser.setEmail(authUser.getEmail());
+            newUser.setDisplayName(authUser.getName());
+            user = createUser(newUser);
+        }
+        // Ensure that the user is a registered SAM user
+        try {
+            samDAO.postRegistrationInfo(authUser);
+        } catch (ConsentConflictException cce) {
+          // no-op in the case of conflicts.
+        } catch (Exception e) {
+            logger.error("Error posting user registration information to Sam for user " + authUser.getName() + ". Error: " + e.getMessage());
+        }
```

We're still swallowing the exception when it would be better handled in the resource class. I would suggest we let all other exceptions bubble up here.

---

Updated this so that we do not catch this exception and instead let this method throw an exception so it is thrown within the resource class.